### PR TITLE
used status gone for inactive promotion and tested

### LINKED ISSUE #472: add new return code on claim attempt in case that promotion is no longer active
return e.g. HTTP 410 Gone

### PR DESCRIPTION
**Pull request by maikelmclauflin**
### Summary

returns an app error so that a 410 (gone) is returned instead of another status code
fixes #472 

### Type of change ( select one )

- [x] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [x] Does your code build cleanly without any errors or warnings?
- [x] Have you used auto closing keywords?
- [x] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: promotion/claim.go [hunk 1]
```diff
@@ -84,6 +86,12 @@ func (service *Service) ClaimPromotionForWallet(
 	if promotion == nil {
 		return nil, errors.New("promotion did not exist")
 	}
+	if !promotion.Active {
```
Comment (evq) [inline promotion/claim.go:89]: can you also cover the case where we've passed the 3 month claim window on the promotion?

Comment (maikelmclauflin) [reply promotion/claim.go]: completed in: ea5f7a4

File: promotion/claim.go [hunk 2]
```diff
@@ -84,6 +90,12 @@ func (service *Service) ClaimPromotionForWallet(
 	if promotion == nil {
 		return nil, errors.New("promotion did not exist")
 	}
+	if !promotion.Active || promotion.ExpiresAt.Before(time.Now()) || time.Since(promotion.CreatedAt) > threeMonths {
```
Comment (husobee) [inline promotion/claim.go:93]:
```go
( promotion.CreatedAt < time.Now().AddDate(0,-3,0) ) 
```
Maybe we could use ^^ instead for the 3 month check.

Comment (maikelmclauflin) [reply promotion/claim.go:93]: that's certainly fine with me. are we thinking this is best for readability?

**Pull request merged**